### PR TITLE
Filter out 10x samples from mixed lanes

### PR DESCRIPTION
**Pull request by mlaszloffy**

### CODE REVIEW

File: workflow-casava/pom.xml [hunk 1]
```diff
@@ -29,5 +29,32 @@
             <version>2.4</version>
         </dependency>
     </dependencies>
-    
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.pyx4j</groupId>
+                <artifactId>maven-junction-plugin</artifactId>
```
Comment (morgantaschuk) [inline workflow-casava/pom.xml]: You put it back?

File: decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java [hunk 2]
```diff
@@ -687,6 +674,24 @@ public void setConfig(Map<String, String> config) {
             }
             LaneProvenance lp = Iterables.getOnlyElement(laneNameToLaneProvenance.get(laneName)).getProvenance();
 
+            //expect one sample provenance provider per lane - or no samples in lane
+            List<ProvenanceWithProvider<SampleProvenance>> sps = laneNameToSampleProvenance.get(laneName);
+
+            if (sps.isEmpty()) {
+                laneValidationErrors.add(String.format("Sample provenance count = [%s], expected 1 or more.", sps.size()));
```
Comment (morgantaschuk) [inline decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java]: Should this `continue`?

File: workflow-casava/pom.xml (hunk 1 above)
Comment (mlaszloffy) [reply workflow-casava/pom.xml]: I have removed this plugin and updated the workflow to use parent pom version 2.4.8 (which does not include maven-junction-plugin).

File: decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java (hunk 2 above)
Comment (mlaszloffy) [reply decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java]: Lane validation errors are collected and returned/printed once all validation rules are applied.

**Pull request merged**